### ISSUE TITLE
Preview images and page descriptions for social media

### ISSUE THREAD

**Issue by ericsteuer** [creativecommons/creativecommons.github.io-source#100]
Hi -- just a suggestion: Blog post pages on the open source blog don't appear to be optimized well for Twitter and Facebook (etc.) to be able to fetch and display image and description previews. See attached screenshots 
<img width="374" alt="Screen Shot 2019-07-18 at 7 29 30 AM" src="https://user-images.githubusercontent.com/8919762/61465967-f0cb7980-a92d-11e9-9875-9b50a6628237.png">

<img width="478" alt="Screen Shot 2019-07-18 at 7 30 53 AM" src="https://user-images.githubusercontent.com/8919762/61466032-09d42a80-a92e-11e9-9b8f-cedf63407c7b.png">

**Comment by SamOdum** [comment]
I can handle this. Please may I proceed?

**Comment by kgodey** [comment]
Sure @SamOdum, I see you've commented on a couple of other issues on this repo as well, please pick one issue at a time so that others have a chance too. Let me know which one you want to tackle first and I'll mark that one as in progress.

**Comment by SamOdum** [comment]
Thank you for responding. May I start with this one?

**Comment by kgodey** [comment]
Go ahead.

**Comment by chimenekouomogne** [comment]
Hi @kgodey ! Pleace may i work with you on this ?

**Comment by kgodey** [comment]
@chimenekouomogne This is already in progress, there is a pull request for it.

**Comment by techievivek** [comment]
@kgodey mam If this is not fixed I will love to work on this one.
Thank you.

**Comment by kgodey** [comment]
@techievivek go ahead

**Comment by techievivek** [comment]
@kgodey hello mam, I tried solving this issue but since all the blog posts need not have image how will we able to show the featured image for same, I think @SamOdum idea is one we can go with.
Or please suggest me something so that I can move forward with or think about it.
The main problem is with the featured image, the description part can be done easily.
Thank you.

**Comment by kgodey** [comment]
We should use the first image on the page/blog post if it exists, otherwise we should not have a featured image at all.

**Comment by techievivek** [comment]
@kgodey so in case if no images are part of blog/page can we use cc logo. So that it looks genuine and better while sharing.

**Comment by kgodey** [comment]
I think it's best to skip the featured image entirely if there's no image in the post/page.

**Comment by techievivek** [comment]
@kgodey hello mam, I have almost worked on this issue but just some queries before I start a pull request.
1. For index page i.e. opensource.creativecommons.org , since I am getting first image by default(if there), so here it is not showing CC logo rather a black & white keyboard image, as that is the first image for this page.
2. Description part of meta tag, I don't know what I have to put here, first, I thought to go with body(content part of post), but that is **markdown content** so we will need to strip it to get text content (there is a plugin too in lektor [striphtmltags](https://www.getlektor.com/plugins/lektor-strip-html-tags/)), but i am unable to install it, secondly we will have to **slice** the content(as post content can be very long) and there is again a plugin in lektor [markdown-excerpt](https://www.getlektor.com/plugins/lektor-markdown-excerpt/), but this is also not working.
Please help me so that i can initiate the PR.
Thank you

**Comment by techievivek** [comment]
@kgodey hello mam, should i create a pull request and then you will check it out?

**Comment by kgodey** [comment]
@techievivek 
1. that's fine
2. what errors are you running into with the Lektor plugins?

**Comment by techievivek** [comment]
@kgodey hello mam, I have added both striphtmltags and markdown-excerpt plugin to the project so that I can convert body content into plain text and then just get the first line content using markdown-excerpt.
I have added by running the command 
```
lektor plugin add markdown-excerpt
lektor plugin add strip-html-tags
```
This is the content of file _.lektorproject_ after I run the above command.
![lektor diff](https://user-images.githubusercontent.com/35863227/73482350-94104780-43c3-11ea-9160-0951beded1c0.png)
But none of them is working.
Below is the error I am receiving.
![error striphtmltags](https://user-images.githubusercontent.com/35863227/73482349-9377b100-43c3-11ea-9124-0249546ed4fc.png)
# Progress till now
I have just added the image functionality, i.e. it will select an image if there.
```
{% set image = this.attachments.images.first() %}
      {% if image %}
<meta property="og:image" content="{{ image | url }}" />
      {% endif %}
```
Please suggest me how do I proceed with this?
Thank you

**Comment by techievivek** [comment]
@kgodey hello mam, please look into this.
Thank you

**Comment by kgodey** [comment]
@techievivek please create a PR so that I can see why the error is happening.

**Comment by techievivek** [comment]
@kgodey mam, I have created a PR, please look into it.
Thank you

**Comment by techievivek** [comment]
@kgodey hello mam, did u have a look at it?

**Comment by kgodey** [comment]
Not yet, it will take me a couple more days, sorry.

**Comment by techievivek** [comment]
@kgodey Ok mam, sure. I am looking at other repo under CC.

**Comment by techievivek** [comment]
@kgodey Mam, can we just show the image (first one in the post) and the title while sharing on different platform.( we can skip the description part).
I am attaching a picture on how it looks while sharing for the above condition.
![Screenshot_20200219-164139](https://user-images.githubusercontent.com/35863227/74829382-f01d1a80-5336-11ea-98fb-98cec5dd157f.jpg)
( The link part will be of the CC site, I am using ngrok to tunnel localhost).
If we can go with this I will update my PR.
Or else we just have to sort the description part somehow.

**Comment by techievivek** [comment]
@kgodey mam please take a look at above suggestion.

**Comment by kgodey** [comment]
@techievivek I think GitHub ate my last comment because I definitely replied to this :/

Yes, we can skip the description for now. Could you make a separate GitHub issue for adding the description once this is merged? We can tackle that separately.

**Comment by techievivek** [comment]
@kgodey Yes mam I will do that. Also please take a look at one of the bug that I experienced today here #182

**Comment by techievivek** [comment]
@kgodey I have created a new PR, please have a look. And I will add a new issue for the description part.

**Comment by techievivek** [comment]
@kgodey I have added a new issue for description part, please have a look and suggest if I have to change anything.
#184
